frame/curve-amm/src/lib.rs

```diff
@@ -834,6 +835,11 @@ pub mod pallet {
 			Ok(pool_id)
 		}
 
+		pub fn get_assets(pool_id: &T::PoolId) -> Option<Vec<T::AssetId>> {
+			let assets_pair = PoolAssets::<T>::get(pool_id)?;
+			Some(vec![assets_pair.0, assets_pair.1])
+		}
```

Comment:
it can rewrite so:
```
pub fn get_assets(pool_id: &T::PoolId) -> Result<Vec<T::AssetId>, DispatchError> {
  let assets_pair = PoolAssets::<T>::get(pool_id).ok_or(Error::<T>::InconsistentStorage)?;
  Ok(vec![assets_pair.0, assets_pair.1])
}
```
then calls of this function will look so:
```
let assets = Self::get_assets(&pool_id)?;
```

Reply:
I don't mind doing that, I just thought to make it similar to other functions surrounding. Note all of them return Option<> and based on that caller of those function returns Error. However it is not required. if @hussein-aitlahcen @dzmitry-lahoda prefer this way then I can update.

Reply:
i am for splitting function into 2.

second, allow to iter over tuples of assets (a,b,c).iter(). you can generalize pair into iterator and than to .collect(). 

for options vs error.  as method in palllet, and pallet has PoolNotFound error, error fits better here.

Reply:
and vec! is allocation, you can create iterator and avoid that. or do not iterate at all without vector.

you can make view onto tuple as iterator (without move) and iterate without allocation. i thin exchange is hot pass and if possible should not allocate.

Reply:
if you need iteration over stack allocated obj, use an array. Recently with const_generics has had loads of improvements.

Reply:
that will work too 

<img width="553" alt="image" src="https://user-images.githubusercontent.com/757125/147685567-ef96ca78-a148-4a30-93aa-c2d257aaa2cf.png">

It will have nicer serde too.

Reply:
(in my pr)